### PR TITLE
Integration test cross section with no proton charge

### PR DESCRIPTION
**Pull request by jmborr**
### Description of work:

- [x] Source added/refactored?
- [x] Added unit tests?
- [x] Added integration tests?

### To Test by Reviewer:
Checkout the PR and run UI tests. It's necessary that `/SNS/REF_M` is mounted on the testing machine
```bash
git fetch origin pull/55/merge:pr55
git switch pr55
conda activate quicknxs  # see conda environment file "environment.yml"
pytest ./test/ui
```
The reviewer should check for the discussed software practices

- [ ] all internal functions should have an underbar, as is python standard
- [ ] need more comments in meat of code explaining intent
- [ ] clearly named variables (better to be verbose in variable names)
- [ ] does the code follow the other [coding standards](https://scse.ornl.gov/drtsans/developer/index.html)?

### CODE REVIEW

File: test/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import os
+
+SNS_REFM_MOUNTED = os.path.isdir("/SNS/REF_M/")
+ONE_SECOND = 1000  # ms, a wait time for qtbot
```
Comment (backmari) [inline test/__init__.py:4]: is this used or can it be removed?

File: test/ui/ui_utilities.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    r"""Set the text of a widget and optionally press enter."""
+    assert getattr(widget, "setText", None) is not None
+    widget.setText(text)
+    if press_enter:
+        widget.setFocus(QtCore.Qt.MouseFocusReason)
+        widget.returnPressed.emit()
+
+
+def data_from_plot1D(widget: "MplWidget", line_number=0) -> tuple:
+    r"""Get the data from an MplWidget representing showing a 1D plot
```
Comment (backmari) [inline test/ui/ui_utilities.py:21]: remove "showing"

File: test/ui/ui_utilities.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    -------
+    X and Y data as a tuple of numpy arrays
+    """
+    figure = widget.canvas.fig
+    axes = figure.get_axes()[0]
+    return axes.get_lines()[line_number].get_data()
+
+
+def data_from_plot2D(widget: "MplWidget") -> MaskedArray:
+    r"""Get the data from an MplWidget representing showing a 1D plot
```
Comment (backmari) [inline test/ui/ui_utilities.py:32]: remove "showing"

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+### To Test by Reviewer:
+
+<!-- Instructions for testing. -->
+
+The reviewer should check for the discussed software practices
+
+- [ ] all internal functions should have an underbar, as is python standard
+- [ ] need more comments in meat of code explaining intent
+- [ ] clearly named variables (better to be verbose in variable names)
+- [ ] does the code follow the other [coding standards](https://scse.ornl.gov/drtsans/developer/index.html)?
```
Comment (backmari) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:16]: I got a "down for maintenance" page when I clicked this link. Is it down temporarily or permanently? (I want to read it :smile_cat:)

**Review by backmari — COMMENTED** [review]
Minor comments. Thanks for the examples of tests with qtbot, I will need that soon...

File: test/__init__.py (hunk 1 above)
Comment (jmborr) [reply test/__init__.py:4]: the purpose is to have it when we have a test containing `qtbot.wait(some_time)`, for instance:
```python
from tests.import ONE_SECOND
MainWindow().show()
qtbot.wait(2 * ONE_SECOND)  # stop execution on the main thread for two seconds
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 4 above)
Comment (jmborr) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:16]: oops, I didn't even now I inserted this. I blinded did a copy/paste from `drtsans` pull request template. I'll get rid of it.

**Review by backmari — APPROVED** [review]

**Pull request merged**